### PR TITLE
Initially set file's basename as dataset name

### LINKED ISSUE #246: Use a file's base name as default dataset name
When opening a file with the GUI for the first time, preset dataset name with the file's base name.

### PR DESCRIPTION
**Pull request by AnandBaburajan**
Fixes #246 

## Contributor Checklist:

* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution

### CODE REVIEW

**Comment by AnandBaburajan** [comment]
Sorry for the multiple commits and issues! I learned a lot about the project's code today. Should I avoid the "similar blocks of code found in multiple locations" issue if I come across the same again? Thanks!

**Comment by sk1p** [comment]
First, don't worry about the code coverage issue - sadly the tool is sometimes flaky and doesn't correctly measure or merge coverage from the different CI jobs.

> Sorry for the multiple commits and issues!

Don't worry, we know how it is :grin: sometimes you push and get issue after issue from the CI checks... 

> I learned a lot about the project's code today.

That's good to hear!

> Should I avoid the "similar blocks of code found in multiple locations" issue if I come across the same again?

I think the way you handled it is perfect: you addressed duplication in the code you added. The remaining issue was there before - and sometimes it is not so easy to prevent duplicate code. If you have a good idea how to abstract some duplication away, you are free to do so of course. But the code has to be at least as readable as it was before.

**Comment by AnandBaburajan** [comment]
> If you have a good idea how to abstract some duplication away, you are free to do so of course.

I'll surely work on it by next weekend when I'm finished with my class tests! :)

**Comment by sk1p** [comment]
Oh, one more thing: I noticed you are working on your master branch - this may get you in trouble in the future, when you want to create a new "clean" branch. It is best practice to always create a new branch for each pull request. There is no need to change this for this PR (just keep it as is until this PR is merged!), but I would suggest to keep this in mind in the future. Let me know if you need help with git!

**Comment by AnandBaburajan** [comment]
> always create a new branch for each pull request

Thanks, noted!

**Comment by AnandBaburajan** [comment]
> libertem-server says "unknown file format" or something, and the GUI displays the "raw" dialog

I'm working on it and I'll check #645 too.

**Comment by sk1p** [comment]
> I'm working on it and I'll check #645 too.

Awesome! For #645, you may need a real DM file - you could download `experimental_data.7z` from https://zenodo.org/record/2566137. Trying to load one of the 'diffpattern_*.dm3' files from the archive should trigger that bug.

**Comment by AnandBaburajan** [comment]
> Awesome! For #645, you may need a real DM file - you could download `experimental_data.7z` from https://zenodo.org/record/2566137. Trying to load one of the 'diffpattern_*.dm3' files from the archive should trigger that bug.

> libertem-server says "unknown file format" or something, and the GUI displays the "raw" dialog

I'll finish working on these tasks by next weekend (class tests).

**Comment by AnandBaburajan** [comment]
> if you open a "raw" file, the name is not properly set

I've a doubt, does that also happen to raw files containing real data? Or just empty raw files? I couldn't test with real raw files because I've only access to files of a few formats as of now.

If the aforementioned issue only affects empty raw files, then the feature I added works for all non-empty and valid files with formats supported by LiberTEM. So, shouldn't I be working on error handling for empty and invalid formats and #645 instead? Thanks for helping out! :)

**Comment by sk1p** [comment]
> I've a doubt, does that also happen to raw files containing real data? Or just empty raw files? I couldn't test with real raw files because I've only access to files of a few formats as of now.

As far as I remember, it also affects "real raw files" (compare the results of the detect network call for a file format that supports parameter detection like HDF5 to the results you get if you try to open a raw file). But the nice thing is, the definition of a "real raw file" is quite lax: you can generate one yourself!

The following should do:

```python
import numpy as np
real_data = np.random.randn(16, 16, 16, 16).astype("float32")
real_data.tofile("/tmp/real_raw_file.raw")
```

You can then load the data via the Python API like this:

```python
from libertem.api import Context
ctx = Context()
ds = ctx.load("raw", path="/tmp/something.raw", scan_size=(16, 16), dtype="float32", detector_size=(16, 16))
```

Or by entering those parameters (scan_size, dtype, detector_size) into the load dialog of the GUI.

**Comment by AnandBaburajan** [comment]
> it also affects "real raw files" (compare the results of the detect network call for a file format that supports parameter detection like HDF5 to the results you get if you try to open a raw file)

Thank you for clarifying and for the "real raw file" trick! I'll work on it asap.

**Comment by AnandBaburajan** [comment]
Sorry for the merge conflict! I'll make sure to keep the repos in sync next time. The updated code sets the initial file name for all files including raw files. :)

File: client/src/dataset/helpers.ts [hunk 1]
```diff
@@ -15,6 +16,10 @@ export function getInitial<T extends object, K extends keyof T, V>(key: K, other
     return res;
 }
 
+export function getInitialName(initial: any, path: string){
```
[the remark this answers is not in the document]
Comment (AnandBaburajan) [reply client/src/dataset/helpers.ts]: > Using `any` completely throws out all of the type safety gained by using TypeScript

Thanks! I'm new to Typescript but I'll do a course on it asap.

**Comment by sk1p** [comment]
Thanks a lot, merged!

**Pull request merged**